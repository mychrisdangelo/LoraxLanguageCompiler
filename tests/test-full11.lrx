// declaretree(t)
// declaretree(s)

// definetree(t)
// defineTree(s)

// t = final_tree;
// s = final_tree;

// t->children

// a = access_tree(t, offset);
// //t->children + offset TYPE struct tree **
// assign_tree(a, s);
// *a = s;


// //rhs is null
// //lhs is null
// //something exists on both sides

// struct child_accessed
// --> child
// --> offset
// --> tree

int main()
{
	tree <int> t(2);
	tree <int> s(2);

	t = 3[4[9[101, 102], 10], 5];
	s = 6[7, 8];

	t%0%0%1 = s;
	print(t, '\n');
	t = s;
	print(t, '\n');

}