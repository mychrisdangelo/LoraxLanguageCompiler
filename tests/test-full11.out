3[4[9[101[null,null],6[7[null,null],8[null,null]]],10[null,null]],5[null,null]]
6[7[null,null],8[null,null]]
